newton/_src/geometry/broad_phase_common.py

```
@@ -59,6 +59,50 @@ def binary_search(values: wp.array(dtype=Any), value: Any, lower: int, upper: in
+    filter_pairs: wp.array(dtype=wp.vec2i, ndim=1),
+    num_filter_pairs: int,
+) -> bool:
+    """Return True if pair is in the sorted filter_pairs array (binary search).
+
+    filter_pairs must be sorted lexicographically (by .x then .y). Each entry
+    should be canonical (min, max). When num_filter_pairs is 0, returns False
+    without accessing the array.
+    """
+    if num_filter_pairs <= 0:
+        return False
+    low = int(0)
```

Ruff (0.14.14)

[warning] Value being cast to `int` is already an integer

Remove unnecessary `int` call

(RUF046)